### PR TITLE
p2p:  Solve thread leak problem &  Send disconnect exception problem

### PR DESCRIPTION
**Pull request by wubin12**
Solve thread leak problem &  Send disconnect exception problem

### CODE REVIEW

File: src/main/java/org/tron/common/overlay/server/MessageQueue.java [hunk 1]
```diff
@@ -126,13 +133,15 @@ public void disconnect(ReasonCode reason) {
   }
 
   private void disconnect(DisconnectMessage msg) {
-    ctx.writeAndFlush(msg);
-    ctx.close();
+    if (ctx != null){
```
[the remark this answers is not in the document]
Comment (wubin12) [reply src/main/java/org/tron/common/overlay/server/MessageQueue.java]: There is an asynchronous process for closing the connection

[the remark this answers is not in the document]
Comment (Rovak) [reply src/main/java/org/tron/common/overlay/server/MessageQueue.java]: Something is calling `close` to early at https://github.com/tronprotocol/java-tron/blob/efbfb1a2ed32b6a92ed1016106ab17f39f02f27c/src/main/java/org/tron/common/overlay/server/MessageQueue.java#L209. The original ethereumj code doesn't call `close` https://github.com/ethereum/ethereumj/blob/develop/ethereumj-core/src/main/java/org/ethereum/net/MessageQueue.java#L176 so maybe closing the connection should be done somewhere else

[the remark this answers is not in the document]
Comment (wubin12) [reply src/main/java/org/tron/common/overlay/server/MessageQueue.java]: Our business logic is not the same as the ethereumj，There is an asynchronous process for closing the connection，If  not have a null pointer judgment,  program will throw an exception

[the remark this answers is not in the document]
Comment (Rovak) [reply src/main/java/org/tron/common/overlay/server/MessageQueue.java]: @wubin12 thanks for the explanation

**Pull request merged**